clevertap-core/src/main/java/com/clevertap/android/sdk/LocalDataStore.java

```diff
@@ -505,8 +524,26 @@ private void persistLocalProfileAsync() {
             @Override
             public void run() {
                 synchronized (PROFILE_FIELDS_IN_THIS_SESSION) {
-                    long status = dbAdapter
-                            .storeUserProfile(profileID, new JSONObject(PROFILE_FIELDS_IN_THIS_SESSION));
+                    HashMap<String, Object> profile = PROFILE_FIELDS_IN_THIS_SESSION;
+
+                    boolean passFlag = true;
+                    // Encrypts only the pii keys before storing to DB
+                    for (String piiKey : piiDBKeys) {
+                        if (profile.get(piiKey) != null) {
+                            Object value = profile.get(piiKey);
+                            String encrypted = cryptHandler.encrypt((String) value, piiKey);
```

Comment:
@Anush-Shand Don't we requirescheck here before casting directly?

Reply:
@piyush-kukadiya name,phone,identity,email will always be a string?

Reply:
It's defined as <String,Object> let's be safe

Reply:
Yess. Sounds better